src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +144,1336 @@ static DamageTable D_80BBF858 = {
+                    play_sound(NA_SE_SY_ERROR);
+                    return 0x673;
+                }
+                this->powderKegPrice = play->msgCtx.unk1206C;
+                if (gSaveContext.save.playerData.rupees < this->powderKegPrice) {
+                    this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                    this->actionFlags |= EN_S_GORO_ACTIONFLAG_TIRED;
+                    play_sound(NA_SE_SY_ERROR);
+                    return 0x674;
+                }
+                if ((gSaveContext.save.day == 3) && (gSaveContext.save.isNight)) {
+                    this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                    func_8019F208();
+                    return 0x676;
+                }
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                func_8019F208();
+                return 0x675;
+            }
+            if ((gSaveContext.save.day == 3) && (gSaveContext.save.isNight)) {
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+                return 0x672;
+            }
+            this->actionFlags |= EN_S_GORO_ACTIONFLAG_LASTMESSAGE;
+            return 0x671;
+    }
+    return 0x0;
+}
+
+/**
+ * Links to actor of type EN_JG (Goron Elder), returns whether operation successful.
+ */
+s32 EnSGoro_SetPartnerActor_GoronElder(EnSGoro* this, PlayState* play) {
+    this->otherGoron = SubS_FindActor(play, this->otherGoron, ACTORCAT_NPC, ACTOR_EN_JG);
+    if (this->otherGoron) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Links to actor of type EN_GK (Goron Elder's Son), returns whether operation successful.
+ */
+s32 EnSGoro_SetPartnerActor_GoronChild(EnSGoro* this, PlayState* play) {
+    this->otherGoron = NULL;
+    this->otherGoron = SubS_FindActor(play, this->otherGoron, ACTORCAT_NPC, ACTOR_EN_GK);
+    if (this->otherGoron) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Looks like it manages the snoring cycle for when rolled up and asleep.
+ */
+void EnSGoro_UpdateSleeping(EnSGoro* this, PlayState* play) {
+
+    s16 curFrame = this->skelAnime.curFrame;
+
+    this->snorePhase += 0x400;
+    this->scaleFactor = Math_SinS(this->snorePhase) * 0.01f * 0.1f;
+    this->actor.scale.y = 0.01f - this->scaleFactor;
+    this->actor.scale.z = 0.01f - this->scaleFactor;
+    this->actor.scale.x = this->scaleFactor + 0.01f;
+
+    if (!(this->actionFlags & EN_S_GORO_ACTIONFLAG_SUPPRESS_SNORE)) {
+        if (this->snorePhase == 0) {
+            if (this->actionFlags & EN_S_GORO_ACTIONFLAG_SNOREPHASE) {
+                this->actionFlags &= ~EN_S_GORO_ACTIONFLAG_SNOREPHASE;
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_GOLON_SNORE2);
+            } else {
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_SNOREPHASE;
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_GOLON_SNORE1);
+            }
+        }
+    } else {
+        if (gSaveContext.save.weekEventReg[22] & 4) {
```

Comment:
```suggestion
    } else if (gSaveContext.save.weekEventReg[22] & 4) {
```
This can be made into an `else if`